src/filter_tree/ft_to_rsq.c

```diff
@@ -588,7 +586,7 @@ RSQNode *FilterTreeToQueryNode
 	// to caller as a single filter tree
 	// this might happen when the value compared against is a runtime value of
 	// none indexable type e.g. array
-	*none_converted_filters = FilterTree_Combine(trees, tree_count);
+	*none_converted_filters = FilterTree_Clone(FilterTree_Combine(trees, tree_count));
```

Comment:
I believe this invalidates a previous comment I had, now that the `none_converted_filters` is a clone `node_by_index_scan` should be OK.